app/models/tenant_quota.rb

```diff
@@ -36,7 +36,9 @@ class TenantQuota < ApplicationRecord
 
   NAMES = QUOTA_BASE.keys.map(&:to_s)
 
-  validates :name, :inclusion => {:in => NAMES}
+  validates :name,
+            :inclusion  => {:in => NAMES},
+            :uniqueness => {:scope => [:name, :tenant_id], :message => "should be unique per tenant"}
```

Comment:
Do you need to specify `:name` in the scope here? It might be good to add a test to verify